framework/src/modules/chain/blocks/chain.js

```diff
@@ -195,11 +224,29 @@ const saveBlock = async (storage, block, tx) => {
 	parsedBlock.previousBlockId = parsedBlock.previousBlock;
 	delete parsedBlock.previousBlock;
 
-	parsedBlock.transactions.map(transaction => {
+	parsedBlock.transactions.forEach(transaction => {
 		transaction.blockId = parsedBlock.id;
 		return transaction;
 	});
 
+	parsedBlock.transactions = parsedBlock.transactions.map(transaction =>
+		transaction.toJSON()
+	);
```

Comment:
We should override the block id after this block 

```
	parsedBlock.transactions.forEach(transaction => {
 		transaction.blockId = parsedBlock.id;		 		
 		return transaction;
 	});

```

Reply:
Why so?

Reply:
Because I am not sure if `transaction.toJSON()` is preserving that custom attribute we assigned just above manually.